lib/Backend/GlobOpt.cpp

```diff
@@ -4704,6 +4704,14 @@ GlobOpt::ValueNumberDst(IR::Instr **pInstr, Value *src1Val, Value *src2Val)
     case Js::OpCode::Coerce_Str:
         AssertMsg(instr->GetDst()->GetValueType().IsString(),
             "Creator of this instruction should have set the type");
+
+        // Due to fall through and the fact that Ld_A only takes one source,
+        // free the other source here.
+        if (instr->GetSrc2())
```

Comment:
we should only FreeSrc2 if we change into a Ld_A (which won't happen if `this->IsLoopPrePass() || src1ValueInfo == nullptr || !src1ValueInfo->IsString()`)